docs/contributing/backend.md

```diff
@@ -0,0 +1,272 @@
+# Contributing to Nixopus Backend
+
+This guide provides detailed instructions for contributing to the Nixopus backend codebase.
+
+## Setup for Backend Development
+
+1. **Prerequisites**
+   - Go version 1.23.6 or newer
+   - PostgreSQL
+   - Docker and Docker Compose (recommended)
+
+2. **Environment Setup**
+
+   ```bash
+   # Clone the repository
+   git clone https://github.com/raghavyuva/nixopus.git
+   cd nixopus
+   
+   # Set up PostgreSQL database
+   createdb nixopus -U postgres
+   createdb nixopus_test -U postgres
+   
+   # Copy environment template
+   cp api/.env.sample api/.env
+   
+   # Install dependencies
+   cd api
+   go mod download
+   ```
+
+3. **Database Migrations**
+
+   ```bash
+   # Run migrations
+   cd api
+   go run migrations/main.go
+   ```
+
+## Project Structure
+
+The backend follows a clean architecture approach:
+
+```
+api/
+├── internal/
+│   ├── features/      # Feature modules
+│   ├── middleware/    # HTTP middleware
+│   ├── config/        # Application configuration
+│   ├── storage/       # Data storage implementation
+│   ├── utils/         # Utility functions
+│   └── types/         # Type definitions
+├── migrations/        # Database migrations
+└── tests/             # Test utilities
+```
+
+## Adding a New Feature
+
+1. **Create a New Branch**
+
+   ```bash
+   git checkout -b feature/your-feature-name
+   ```
+
+2. **Implement Your Feature**
+
+   Create a new directory under `api/internal/features/` with the following structure:
+
+   ```
+   api/internal/features/your-feature/
+   ├── controller.go   # HTTP handlers
+   ├── service.go      # Business logic
+   ├── storage.go      # Data access
+   └── types.go        # Type definitions
+   ```
+
+   Here's a sample implementation:
```

Comment:
this sample implementation does not properly follows the pattern we are having, let's recheck this

Reply:
Fixed this